osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/periodicmeterreads/ParametersObject.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.periodicmeterreads;
+
+import java.util.List;
+
+import org.joda.time.DateTime;
+import org.openmuc.jdlms.GetResult;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ChannelDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.PeriodTypeDto;
+
+public class ParametersObject {
+    private final PeriodTypeDto periodType;
+    private final DateTime beginDateTime;
+    private final DateTime endDateTime;
+    private final List<DataObject> bufferedObjects;
+    private final ChannelDto channel;
+    private final boolean isSelectiveAccessSupported;
+    private final List<GetResult> results;
+
+    public ParametersObject(final PeriodTypeDto periodType, final DateTime beginDateTime, final DateTime endDateTime,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 7 parameters, which is greater than 6 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00107)